static/solo/js/menu.js

```diff
@@ -0,0 +1,169 @@
+                                '<svg class="tvp-video-play" viewBox="0 0 200 200" alt="Play video">' +
+                                    '<polygon points="70, 55 70, 145 145, 100"></polygon>' +
+                                '</svg>' +
+                            '</div>' +
+                            '<div class="tvp-video-details">' +
+                                '<p class="tvp-video-title">{title}</p>' +
+                                '<p class="tvp-video-duration">-{asset.prettyDuration}</p>' +
+                            '</div>' +
+                        '</div>',
+
+    isset = function(o, p) {
+      var val = o;
+      if (p) val = o[p];
+      return 'undefined' !== typeof val;
+    },
+
+    isEmpty = function(obj) {
+        for(var key in obj) { if (obj.hasOwnProperty(key)) return false;}
+        return true;
+    },
+
+    tmpl = function(template, data) {
+        if (!template && 'object' !== typeof data) return;
+        return template.replace(/\{([\w\.]*)\}/g, function(str, key) {
+            var keys = key.split("."),
+            v = data[keys.shift()];
+            for (var i = 0, l = keys.length; i < l; i++) v = v[keys[i]];
+            return (typeof v !== "undefined" && v !== null) ? v : "";
+        });
+    },
+
+    trimText = function(text, limit) {
+      var t = text || '';
+      var l = limit ? Number(limit) : 0;
+      if (text.length > l) {
+        t = t.substring(0, Number(l)) + '...';
+      }
+      return t;
+    },
+
+    assety = function(data, options){
+      var assets = [];
+      for (var i = 0; i < data.length; i++) {
+        var video = data[i];
+        
+        if (isEmpty(video)) break;
+
+        var asset = video.asset;
+        asset.assetId = video.id;
+        asset.assetTitle = video.title;
+        asset.loginId = video.loginId;
+
+        if (isset(video,'events') && video.events.length) {
+          asset.analyticsLogUrl = video.analytics;
+          asset.analyticsObj = video.events[1].data;
+        } else {
+          asset.analyticsObj = {
+            pg: isset(video,'parentId') ? video.parentId : ( isset(options,'channel') ? options.channel.id : 0 ),
+            vd: video.id, 
+            li: video.loginId
+          };
+        }
+
+        if (!asset.sources) asset.sources = [{ file: asset.videoId }];
+        asset.type = asset.type || 'youtube';
+        assets.push(asset); 
+      }
+      return assets;
+    };
+
+
+  function Menu(player, settings) {
+
+    var that = this;
+    this.allVideos = [];
+
+    this.render = function() {
+      var playlist = settings.data || [];
+      if (playlist.length < 1) return;
+
+      that.menuFrag = document.createDocumentFragment();
+      that.slideMenu = document.createElement('div');
+      that.slideMenu.setAttribute('id', 'tvp-slide-menu');
+      that.slideMenu.innerHTML = menuTemplate;
+      that.hiddenMenu = that.slideMenu.querySelectorAll('#tvp-hidden-menu')[0];
```

Comment:
Isn't better to use `that.slideMenu.getElementById` ?
If this does not work while being used from a node then you can try using `that.slideMenu.querySelector('#tvp-hidden-menu')` this will return the first match, removing the need of doing `[0]`